### PR TITLE
parser: Fix the bug of `alter table table_options, other_alter_specification`

### LINKED ISSUE #4927: The parser.y not support `alter table table_name engine=``,add column column_name type;`
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
when i go to execute `alter table foo engine=``,add column id int;`
RDS support `alter table table_name engine=``;` 

2. What did you expect to see?
tidb can parser the sql

3. What did you see instead?
```
error while parsing sql:
 alter table bar engine=``,add column aaa int, err:line 1 column 29 near " column aaa int" (total length 44)
```

the reason is the `parser.y` not support, 

4. What version of TiDB are you using (`tidb-server -V`)?
```
Git Commit Hash: a0017eda04a1d48e9ec088457afe279a8cd064f4
UTC Build Time:  2017-08-18 10:35:56
```

### PR DESCRIPTION
**Pull request by zimulala**
Fix #4927

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @coocood @tiancaiamao

**Comment by zz-jason** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
@zimulala 
Seems we need to define a new TableOptList that is separated only by space for alter table statement. Then restore the `lowerThanComma` precedence.

The following is copied from the MySQL parser.
```
create_table_options_space_separated:
          create_table_option
        | create_table_option create_table_options_space_separated
```

**Comment by winkyao** [comment]
agree with @coocood

**Comment by zimulala** [comment]
@coocood @winkyao 
Alter table statement also can separate by comma. I tried on MySQL.
PTAL

File: parser/parser_test.go [hunk 1]
```diff
@@ -1499,6 +1500,11 @@ func (s *testParserSuite) TestDDL(c *C) {
 		{"ALTER TABLE t ENGINE = ''", true},
 		{"ALTER TABLE t ENGINE = 'innodb'", true},
 		{"ALTER TABLE t ENGINE = innodb", true},
+		{"ALTER TABLE t ADD COLUMN a SMALLINT UNSIGNED, ADD COLUMN a SMALLINT", true},
+		{"ALTER TABLE t ADD COLUMN a SMALLINT, ENGINE = '', default COLLATE = utf8_general_ci", true},
+		{"ALTER TABLE t ENGINE = '', COMMENT='', default COLLATE = utf8_general_ci", true},
+		{"ALTER TABLE t ENGINE = '', ADD COLUMN a SMALLINT", true},
+		{"ALTER TABLE t default COLLATE = utf8_general_ci, ENGINE = '', ADD COLUMN a SMALLINT", true},
```
Comment (holys) [inline parser/parser_test.go]: How about adding a case  ALTER TABLE `foo`.`bar` ENGINE=``     (engine value in back-quote)

Comment (zimulala) [reply parser/parser_test.go]: Done.

**Pull request merged**